examples/sqlite_drizzle/src/entities/posts/db.ts

```diff
@@ -0,0 +1,134 @@
+import { desc, eq } from 'drizzle-orm';
+import { integer, sqliteTable, text } from 'drizzle-orm/sqlite-core';
+
+import { DrizzleDb } from '../../db';
+import { User, Users } from '../users/db';
+
+export const Posts = sqliteTable('posts', {
+    id: integer('id').primaryKey(),
+    title: text('title').notNull(),
+    body: text('body').notNull(),
+    user_id: integer('user_id')
+        .notNull()
+        .references(() => Users.id)
+});
+export type Post = Omit<typeof Posts.$inferSelect, 'user_id'> & { user: User };
+export type PostCreate = Omit<typeof Posts.$inferInsert, 'id'>;
+export type PostUpdate = Pick<Post, 'id'> & Partial<PostCreate>;
+
+export async function getPosts(
+    db: DrizzleDb,
+    limit: number,
+    offset: number
+): Promise<Post[]> {
+    const results = await db
+        .select()
+        .from(Posts)
+        .innerJoin(Users, eq(Posts.user_id, Users.id))
+        .limit(limit)
+        .offset(offset);
+
+    return results.map((result) => {
+        return {
+            id: result.posts.id,
+            title: result.posts.title,
+            body: result.posts.body,
+            user: result.users
+        };
+    });
+}
+
+export async function getPost(
+    drizzleDb: DrizzleDb,
+    id: number
+): Promise<Post | null> {
+    const results = await drizzleDb
+        .select()
+        .from(Posts)
+        .innerJoin(Users, eq(Posts.user_id, Users.id))
+        .where(eq(Posts.id, id));
+
+    return results.length === 0
+        ? null
+        : {
+              id: results[0].posts.id,
+              title: results[0].posts.title,
+              body: results[0].posts.body,
+              user: results[0].users
+          };
+}
+
+export async function countPosts(drizzleDb: DrizzleDb): Promise<number> {
+    const results = await drizzleDb
```

Comment:
drizzle has a way to use the SQL COUNT function
```javascript    
    const results = await drizzleDb
        .select({
            count: count()
        })
        .from(Posts);

    return results[0]?.count ?? 0;
```

This would be more robust because if you delete posts won't it add post starting at the biggest index and not fill in empty indexes?

Reply:
Resolved